kimchi/src/tests/keccak.rs

```diff
@@ -0,0 +1,119 @@
+use crate::circuits::{
+    constraints::ConstraintSystem, gate::CircuitGate, polynomials::keccak::witness::create,
+    wires::Wire,
+};
+
+use ark_ec::AffineCurve;
+use mina_curves::pasta::{Fp, Pallas, Vesta};
+use rand::Rng;
+
+type PallasField = <Pallas as AffineCurve>::BaseField;
+
+fn create_test_constraint_system() -> ConstraintSystem<Fp> {
+    let (mut next_row, mut gates) = CircuitGate::<Fp>::create_keccak(0);
+
+    // Temporary workaround for lookup-table/domain-size issue
+    for _ in 0..(1 << 13) {
+        gates.push(CircuitGate::zero(Wire::new(next_row)));
+        next_row += 1;
+    }
+
+    ConstraintSystem::create(gates).build().unwrap()
+}
+
+#[test]
+// Test a XOR of 64bit whose output is all ones with alternating inputs
+fn test_xor64_alternating() {
+    let cs = create_test_constraint_system();
+
+    let zero_ones: u64 = 6510615555426900570;
+    let one_zeros: u64 = 11936128518282651045;
+    let witness = create(zero_ones, one_zeros);
+
+    for row in 0..=5 {
```

Comment:
Nit: this seems common to all the tests, worth factoring it into a helper?

Reply:
Done!